mondrian/src/main/java/mondrian/olap/Util.java

```diff
@@ -4402,6 +4404,33 @@ public int match(byte[] a) {
         return map;
     }
 
+    /**
+     * Retrieves the maximum number of constraints in a single 'IN' SQL clause based on dialect information
+     * This value dependes on database products or on their runtime settings.
+     *
+     * @param evaluator The rolap evaluator which contains the DB information
+     * @return The maximum number of constraints in a single 'IN' SQL clause
+     */
+    public static int getMaxConstraintsByProduct(RolapEvaluator evaluator){
+
+        final Dialect dialect = evaluator.getDialect();
+
+        if(dialect == null || dialect.getDatabaseProduct() == null){
+            return MondrianProperties.instance().MaxConstraints.get();
+        }
+
+        DatabaseProduct dbProduct = evaluator.getDialect().getDatabaseProduct();
+
+        switch ( dbProduct ) {
+            case ORACLE:
```

Comment:
Also note that the dialect has some related functions which we should consider.

https://github.com/pentaho/mondrian/blob/master/mondrian/src/main/java/mondrian/spi/Dialect.java#L468

Reply:
There is also a Mondrian property we should consider. How will this change choose between what's set in properties, possibly by the user, versus those hard-coded values?

https://github.com/pentaho/mondrian/blob/master/mondrian/src/main/java/mondrian/olap/MondrianProperties.xml#L870

Reply:
Thanks for your comments.
That was the old implementation and to be honest I assumed the *MaxConstraints* value on the properties file as already hard-coded one i.e. it could be changed by end user but always in accordance to the used database. Is it supposed to be changed to some random values?

Nevertheless I believe my solution can obsolete it, so I thought on another solution and it would be just the increase of the default of that particular value to the maximum possible (i.e. 10000), however I am afraid about some negative impact.

So my vote goes to your first suggestion (i.e. create a method in Dialect, add a default value in JdbcDialectImpl, and override the value by vendor in their respective dialect). What do you think?